### PR TITLE
[SDL] Bump versions (help wanted)

### PR DESCRIPTION
**Pull request by Gnimuc**
- SDL2: 2.0.12 => 2.0.20
- SDL2_ttf: 2.0.15 => 2.0.18

Also switch from ArchiveSource to GitSource.

### CODE REVIEW

**Comment by Gnimuc** [comment]
Looks like the build scripts for SDL_xxx are not trivial to be fixed. I don't have time to dive into the details, so waiting for others to pick this up.

**Comment by giordano** [comment]
Not sure switching to Git sources was a good idea, now Windows builds fail with
```
[05:06:38] configure: error: 
[05:06:38] *** When building from a git clone you should configure and build in a
[05:06:38]     separate directory so you don't clobber SDL_config.h, SDL_revision.h
```

**Comment by giordano** [comment]
And SDL2_image is successful if using the release tarball.

**Comment by giordano** [comment]
And also for SDL2_mixer using the release tarball gets us past the bootstrap failure.  However during the build I get a different issue:
```
[18:58:15] grep: /workspace/destdir/lib/libogg.la: No such file or directory
[18:58:15] sed: /workspace/destdir/lib/libogg.la: No such file or directory
[18:58:15] libtool: link: `/workspace/destdir/lib/libogg.la' is not a valid libtool archive
[18:58:15] make: *** [Makefile:65: build/libSDL2_mixer.la] Error 1
```
@fingolfin we're getting again this issue, again with libogg.

**Comment by giordano** [comment]
Side note: having a single PR which builds multiple package isn't a good idea either.  It makes working through warnings/errors rather cumbersome

**Comment by giordano** [comment]
Another side effect of having a single PR building multiple packages is that now SDL2 works fine (after switching back to the release tarball instead of the git source) but we can't merge it.

I guess there is enough evidence that using the git source was a bad idea.

File: S/SDL2_ttf/build_tarballs.jl
Comment (giordano) [inline S/SDL2_ttf/build_tarballs.jl:10]: You should got back to the release tarball also here, and refresh the patches

File: S/SDL2_ttf/build_tarballs.jl
Comment (giordano) [inline S/SDL2_ttf/build_tarballs.jl:14]: Note that this line must be removed

File: S/SDL2_ttf/build_tarballs.jl
Comment (giordano) [inline S/SDL2_ttf/build_tarballs.jl:18]:
```suggestion
cd $WORKSPACE/srcdir/SDL2_ttf-*/
```

**Review by giordano — CHANGES_REQUESTED** [review]

**Comment by fingolfin** [comment]
> And also for SDL2_mixer using the release tarball gets us past the bootstrap failure. However during the build I get a different issue:
> 
> ```
> [18:58:15] grep: /workspace/destdir/lib/libogg.la: No such file or directory
> [18:58:15] sed: /workspace/destdir/lib/libogg.la: No such file or directory
> [18:58:15] libtool: link: `/workspace/destdir/lib/libogg.la' is not a valid libtool archive
> [18:58:15] make: *** [Makefile:65: build/libSDL2_mixer.la] Error 1
> ```
> 
> @fingolfin we're getting again this issue, again with libogg.

There is no direct reference to libogg in `SDL2_mixer`, so it must be indirect. And indeed, mpg123 references libogg.la in its artifacts, and was last rebuilt November 2019. I'll submit an update for it now.

File: S/SDL2_ttf/build_tarballs.jl
Comment (fingolfin) [inline S/SDL2_ttf/build_tarballs.jl:18]: Maybe
```suggestion
cd $WORKSPACE/srcdir/SDL2_ttf*/
```
so it'll work with both git clones and release tarballs.

**Comment by giordano** [comment]
After #4307, #4308, and #4309 now also SDL2_mixer is working.  Thanks Max for the tip.

Yet another issue with bundling multiple packages in a single PR is that SDL2_* packages depend on SDL2 for the new platforms (aarch64-apple-darwin and armv6l).  So we can't possible merge this PR as is.

**Comment by Gnimuc** [comment]
Thanks for all the help! I opened another PR #4310 to update SDL2.

**Comment by giordano** [comment]
SDL2_mixer is good to go, but once again, being together with other packages which are failing makes it impossible to merge it alone.  There are failures for macOS in SDL2_image and you still need to refresh the patches for SDL2_ttf.  I don't think I'm going to do anything more here.

**Comment by Gnimuc** [comment]
I can successfully build SDL2_image's lastest git main branch for x86_64-apple-darwin14 using its CMake build. But it looks like the M1 support is still in progress(https://github.com/libsdl-org/SDL_image/issues/182#issuecomment-909488050). I think we can try this again when new versions are released in the upstream. At this moment, I only need to update SDL2 to fix https://github.com/JuliaMultimedia/SimpleDirectMediaLayer.jl/issues/60.

**Pull request closed without merging**
